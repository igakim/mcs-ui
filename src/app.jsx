import React from 'react';
import { render } from 'react-dom';
import { Provider } from 'react-redux';

import App from './view/Home';

export default (store) => {
  render(
    <Provider store={store}>
      <App />
    </Provider>,
    document.getElementById('root'),
  );
};
