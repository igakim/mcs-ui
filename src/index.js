import '@babel/polyfill';

// Import Components:
import Button from './components/button/Button';
import Menu from './components/menu/Menu';
import Section from './components/section/Section';
import Card from './components/card/Card';
import Form from './components/form/Form';

export {
  Button,
  Menu,
  Section,
  Card,
  Form,
};
