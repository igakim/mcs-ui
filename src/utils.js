export const isEqualTypes = (type, compareType) => (
  type && type === compareType
);

export const getButtonType = type => (
  type && `mcs-button-${type}`
);

export const getClassNameFromArray = classesList => classesList
  .filter(el => (el !== undefined && el !== null))
  .join('-');

export const isClassComponent = component => (!!(
  typeof component === 'function'
    && component.prototype
    && component.prototype.isReactComponent
));

export const isFunctionComponent = component => (!!((
  typeof component === 'function'
    && !(
      component.prototype
      && component.prototype.isReactComponent
    )
)));

export const isReactComponent = component => (!!((
  isClassComponent(component)
    || isFunctionComponent(component)
)));
