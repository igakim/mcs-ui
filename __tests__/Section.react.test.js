import React from 'react';
import renderer from 'react-test-renderer';
import { Section } from '../src';

test('component should match snapshot', () => {
  const { SectionCell } = Section;

  const component = renderer.create(
    <Section>
      <SectionCell xs={12} sm={12} md={6} lg={6} xl={4}>
        Some super text
      </SectionCell>
      <SectionCell xs={12} sm={12} md={6} lg={6} xl={4}>
        Some other text
      </SectionCell>
      <SectionCell xs={12} sm={12} md={12} lg={12} xl={4}>
        Some long long long (no) text
      </SectionCell>
    </Section>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('component should match snapshot', () => {
  const { SectionCell } = Section;

  const component = renderer.create(
    <Section className="center">
      <SectionCell xs={12} sm={12} md={6} lg={6} xl={4}>
        Some super text
      </SectionCell>
      <SectionCell xs={12} sm={12} md={6} lg={6} xl={4}>
        Some other text
      </SectionCell>
      <SectionCell xs={12} sm={12} md={12} lg={12} xl={4}>
        Some long long long (no) text
      </SectionCell>
    </Section>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});