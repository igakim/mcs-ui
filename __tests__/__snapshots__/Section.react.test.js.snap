// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component should match snapshot 1`] = `
<div
  className="mcs-section "
>
  <div
    className="mcs-section-cell xs-12 sm-12 md-6 lg-6 xl-4 "
  >
    Some super text
  </div>
  <div
    className="mcs-section-cell xs-12 sm-12 md-6 lg-6 xl-4 "
  >
    Some other text
  </div>
  <div
    className="mcs-section-cell xs-12 sm-12 md-12 lg-12 xl-4 "
  >
    Some long long long (no) text
  </div>
</div>
`;

exports[`component should match snapshot 2`] = `
<div
  className="mcs-section center"
>
  <div
    className="mcs-section-cell xs-12 sm-12 md-6 lg-6 xl-4 "
  >
    Some super text
  </div>
  <div
    className="mcs-section-cell xs-12 sm-12 md-6 lg-6 xl-4 "
  >
    Some other text
  </div>
  <div
    className="mcs-section-cell xs-12 sm-12 md-12 lg-12 xl-4 "
  >
    Some long long long (no) text
  </div>
</div>
`;
